systemd_ctypes/event.py

```diff
@@ -84,8 +84,23 @@ def new_event_loop(self):
 def run_async(main):
     asyncio.set_event_loop_policy(EventLoopPolicy())
 
-    if sys.version_info >= (3, 7, 0):
-        asyncio.run(main)
-    else:
+    polyfill = sys.version_info < (3, 7, 0) and not hasattr(asyncio, 'run')
+    if polyfill:
+        # Polyfills for Python 3.6:
         loop = asyncio.get_event_loop()
-        loop.run_until_complete(main)
+
+        assert not hasattr(asyncio, 'get_running_loop')
+        asyncio.get_running_loop = lambda: loop
+
+        assert not hasattr(asyncio, 'create_task')
+        asyncio.create_task = loop.create_task
+
+        assert not hasattr(asyncio, 'run')
+        asyncio.run = loop.run_until_complete
+
+        asyncio._systemd_ctypes_polyfills = True
+
+    asyncio.run(main)
+
+    if polyfill:
+        del asyncio.create_task, asyncio.get_running_loop, asyncio.run
```

Comment:
Why? What's wrong with adding these polyfills in the library initializer, and keeping them there? Then you don't penalize every run_async() call, and they really don't hurt.

Reply:
The issue is that I don't have the loop initialized at that point.

We could add a simple version of them which queries the loop using `get_current_loop()`, but that function has an awful lot of caveats (which is exactly why we're supposed to avoid using it in new code).

But there's also one more issue: as a general rule, I don't like python modules which introduce side-effects by importing them.